cocos/3d/lights/directional-light-component.ts

```diff
@@ -111,6 +111,20 @@ export class DirectionalLight extends Light {
         }
     }
 
+    /**
+     * @en Visibility mask of the light, declaring a set of node layers that will be visible to this light(directional has not).
+     * @zh 光照的可见性掩码，声明在当前光照中可见的节点层级集合（该功能不包括方向光的设置）。
+     * @internal
+     * @deprecated
+     */
+    @visible(false)
+    set visibility (vis: number) {
+        this._visibility = vis;
+    }
+    get visibility (): number {
+        return this._visibility;
+    }
```

Comment:
Why add new deprecated functions?

Reply:
This is a method of the base class, I overload it on the directional light component because I want to mask it in the editor display screen. At the moment we don't have a good implementation of visibility in the directional light, so we are blocking this for now. We marked it as @internal @deprecated because we don't want external developers and internal developers to use this interface, and when we support directional light visibility in the future, we can just remove this interface

Translated with www.DeepL.com/Translator (free version)

Reply:
In unity, all lighting components support visibility control, so we should also implement this visibility control in the base class

Reply:
If so, i think mark as `@engineInternal` is enough.